test/subspace-test-runtime/src/lib.rs

```diff
@@ -355,10 +356,9 @@ impl pallet_messenger::HoldIdentifier<Runtime> for HoldIdentifier {
 }
 
 impl VariantCount for HoldIdentifier {
-    // TODO: HACK this is not the actual variant count but it is required see
-    // https://github.com/autonomys/subspace/issues/2674 for more details. It
-    // will be resolved as https://github.com/paritytech/polkadot-sdk/issues/4033.
-    const VARIANT_COUNT: u32 = 10;
+    const VARIANT_COUNT: u32 = 1
+        + mem::variant_count::<DomainsHoldIdentifier>() as u32
+        + mem::variant_count::<MessengerHoldIdentifier>() as u32;
```

Comment:
It is still a bit awkwards that this doesn't use `mem::variant_count::<Self>() as u32;`.
What about flattening everything in `HoldIdentifier`?